docs/SGX Static Binary Demo Documentation.md

```diff
@@ -0,0 +1,25 @@
+## Intel SGX 2 Static Binary Demo
+
+This demo is designed to illustrate a static binary running in an SGX Enclave. This demo does not include attestation.
+
+### Hardware Requirements
+SGX2 support is required to run this demo. Check with your hardware manufacturer to determine if your hardware has SGX2 support enabled. Alternatively, you may reference the linked Wiki page for an incomplete but growing list of SGX enabled hardware.
+https://github.com/ayeks/SGX-hardware/#hardware-with-sgx2-support
```

Comment:
We ship a tool called `sgx-show` in the Enarx repo which can report on SGX support for the platform the program is being ran on.

I wonder if we should supplement these instructions by showing how to build and run `sgx-show` if not replace them entirely in favor of `sgx-show`.

Anyone have any thoughts on this?

Reply:
Though I think linking to the hardware page is also useful. So IMO we should have both.

Reply:
> We ship a tool called `sgx-show` in the Enarx repo which can report on SGX support for the platform the program is being ran on.
> 
> I wonder if we should supplement these instructions by showing how to build and run `sgx-show` if not replace them entirely in favor of `sgx-show`.
> 
> Anyone have any thoughts on this?

Should this be detailed within this document or a link to a related Wiki page?

Reply:
> Should this be detailed within this document or a link to a related Wiki page?

Let's detail it here. Here's what I'm proposing we should morph the last paragraph into:

Once you have completed the installation instructions, confirm the environment is configured correctly for running the demo. From the root of the Enarx repository:

```shell
$ cd sgx-show
$ cargo run
```

Note that an appropriate environment will have checkmarks next to all entries that `sgx-show` outputs.

Finally, run the demo. From the root of the Enarx repository:

```shell
$ cargo make build
$ enarx-keep-sgx/target/debug/enarx-keep-sgx \
  --code payload/target/x86_64-unknown-linux-musl/debug/payload \
  --shim enarx-keep-sgx-shim/target/x86_64-unknown-linux-musl/debug/enarx-keep-sgx-shim
```

Reply:
> > Should this be detailed within this document or a link to a related Wiki page?
> 
> Let's detail it here. Here's what I'm proposing we should morph the last paragraph into:
> 
> Once you have completed the installation instructions, confirm the environment is configured correctly for running the demo. From the root of the Enarx repository:
> 
> ```shell
> $ cd sgx-show
> $ cargo run
> ```
> […]

You might want to move the `Rust` chapter on top of this, then